backend/routes/quests.go

```diff
@@ -211,6 +220,58 @@ func GetMainUserQuests(w http.ResponseWriter, r *http.Request) {
 	routeutils.WriteDataJson(w, string(quests))
 }
 
+func GetDailyQuestStatusProgress(w http.ResponseWriter, r *http.Request) {
+	userAddress := r.URL.Query().Get("address")
+	if userAddress == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing address parameter")
+		return
+	}
+
+	dayIndexStr := r.URL.Query().Get("dayIndex")
+	if dayIndexStr == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing dayIndex parameter")
+		return
+	}
+
+	dayIndex, err := strconv.Atoi(dayIndexStr)
+	if err != nil {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Invalid dayIndex parameter")
+		return
+	}
+
+	quests, err := core.PostgresQueryJson[QuestProgress](
+		`SELECT quest_id AS QuestId, 0 AS Progress, 100 AS Needed, quest_type as QuestType
```

Comment:
There are no progress and needed fields that will be queried here, just the id and type.
The progress will be computed after.

Then you will just return the quest id, progress, and needed values like in the spec